services/QuillLMS/app/services/google_integration/refresh_access_token.rb

```diff
@@ -71,7 +71,7 @@ def refresh
     private def parse_attributes(data)
       {}.tap do |attributes|
         attributes[:access_token] = data['access_token'] if data['access_token'].present?
-        attributes[:expires_at] = data['expires_in'] if data['expires_in'].present?
+        attributes[:expires_at] = data['expires_in'].seconds.from_now if data['expires_in'].is_a?(Integer)
```

Comment:
Is there a non-Integer value we should handle here?  (That is, could this value be `present?` but not an Integer?, and do we care if that's the case?)

Reply:
Good question.  I had to check, but [defined](https://www.oauth.com/oauth2-servers/making-authenticated-requests/refreshing-an-access-token/) to be the number of seconds.

Reply:
![Screen Shot 2022-10-17 at 3 47 08 PM](https://user-images.githubusercontent.com/2057805/196268701-e0c35f1e-73f2-477d-acca-ba77f3599810.png)

Reply:
In that case, do we even need to use `is_a?`?  I don't have a strong preference, to be honest, I just noticed we were creating a specific condition and not handling other cases.  There's probably not practical difference for us between `is_a?` and `present?` here.

Reply:
The [spec](https://github.com/empirical-org/Empirical-Core/pull/9847/files#diff-e95fea07468bd8b1a33ff1c01cee661810f8d9731c2f395b5dc37455e8432b72L97) had a passing in of a string 'foo'.  Putting a `present?` would permit that, but then would trigger an error with the `data['expires_in'].seconds.from_now` call.  Using is_a?(Integer) ensures the `.seconds` call is made with the correct type.  I would agree it's redundant for the API, but I think it's more transparent for testing.